cloud/gcp/deploy/sql.go

```diff
@@ -17,13 +17,189 @@
 package deploy
 
 import (
+	"context"
 	"fmt"
+	"strings"
+	"time"
 
+	cloudbuild "cloud.google.com/go/cloudbuild/apiv1/v2"
+	"cloud.google.com/go/cloudbuild/apiv1/v2/cloudbuildpb"
+	"github.com/avast/retry-go"
+	"github.com/nitrictech/nitric/cloud/common/deploy/image"
 	deploymentspb "github.com/nitrictech/nitric/core/pkg/proto/deployments/v1"
+	"github.com/pulumi/pulumi-docker/sdk/v4/go/docker"
+	"github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/artifactregistry"
+	"github.com/pulumi/pulumi-gcp/sdk/v6/go/gcp/sql"
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
 )
 
-// TODO
+type CloudBuild struct {
+	pulumi.ResourceState
+
+	ID pulumi.StringOutput
+}
+
 func (a *NitricGcpPulumiProvider) SqlDatabase(ctx *pulumi.Context, parent pulumi.Resource, name string, config *deploymentspb.SqlDatabase) error {
-	return fmt.Errorf("Sql databases are unimplemented in the nitric GCP provider")
+	_, err := sql.NewDatabase(ctx, name, &sql.DatabaseArgs{
+		Name:           pulumi.String(name),
+		Instance:       a.masterDb.Name,
+		DeletionPolicy: pulumi.String(a.GcpConfig.Databases[name].DeletionPolicy),
```

Comment:
I seem to be getting a nil pointer error here.

```
iagnostics:
  pulumi:pulumi:Stack (nitric-prisma-final-nitric-prisma-final-gcpsql):
    error: an unhandled error occurred: 1 error occurred:
        * recovered panic: runtime error: invalid memory address or nil pointer dereference
     Stack: goroutine 90 [running]:
    runtime/debug.Stack()
        C:/Program Files/Go/src/runtime/debug/stack.go:24 +0x5e
    github.com/nitrictech/nitric/cloud/common/deploy/provider.(*PulumiProviderServer).Up.createPulumiProgramForNitricProvider.func2.1()
        C:/projects/nitric/nitric/cloud/common/deploy/provider/pulumi.go:80 +0x3d
    panic({0x2fb66a0?, 0x86e6000?})
        C:/Program Files/Go/src/runtime/panic.go:770 +0x132
    github.com/nitrictech/nitric/cloud/gcp/deploy.(*NitricGcpPulumiProvider).SqlDatabase(0xc00034a9c0, 0xc0007c5440, {0x43f1380, 0xc0007fcdd0},
{0xc0001aa070, 0x5}, 0xc0005deac0)
        C:/projects/nitric/nitric/cloud/gcp/deploy/sql.go:48 +0x1ff
```

Reply:
Suggestion, check for a.GcpConfig.Databases[name] != nil then apply the deletion policy.